### PR TITLE
[Task/cra 110] Bugs fixed and added test cases

### PR DESCRIPTION
**Pull request by Riya-Singhal15**
Some bugs are fixed like -

- Get different error response (result not found and code not found) for getting code not found error and result not present in database when getting result 

- Get a invalid token response whenever token is mismatch.

And also added some test case for the Admin, tpo, student and college API

### CODE REVIEW

File: backend/test/AdminAPI.test.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
+// "use strict";
```
Comment (kanchan-einn) [inline backend/test/AdminAPI.test.js:1]: If this file is not in use, please delete it.

File: backend/package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "test": "mocha --watch --timeout 5000 ",
-    "start": "pm2 start server.js"
+    "start": "nodemon server.js"
```
Comment (kanchan-einn) [inline backend/package.json:8]: revert your changes to this file.

File: backend/controller/errorHandling.js [hunk 3]
```diff
@@ -183,6 +183,28 @@ function idNotFoundErrorHandler() {
...
+    server_msg: "No file found from user, So can't get the response",
+    server_error_ref: Date.now() + randomGenerate(),
+  };
+  return err;
+}
+
+function validEmailNotFoundErrorHandler() {
+  const err = {
+    status: Constants.er_not_found, //404
+    message: "Please provide a valid email with the xlxs file",
```
Comment (kanchan-einn) [inline backend/controller/errorHandling.js:200]: What is the relation between email and xlxs file?

File: backend/controller/errorHandling.js [hunk 4]
```diff
@@ -276,9 +309,9 @@ function questionPaperExistErrorHandler() {
 function studentExistErrorHandler() {
   const err = {
     status: Constants.er_failure, //400
-    message: "Result is already present in the database for the particular student id",
+    message: "You already submitted the exam",
     error_info: "Error",
-    server_msg: "Result already exist in the database, So can't add the same result twice",
+    server_msg: "Already submitted the exam and result exist in the database, So can't add the same result twice",
```
Comment (kanchan-einn) [inline backend/controller/errorHandling.js:314]: Change message to "You have already submitted the result for this exam".

File: backend/test/AdminModel.test.js [hunk 5]
```diff
@@ -1,126 +1,126 @@
-const mongoose = require("mongoose");
-const Admins = require("../model/Admin");
-const assert = require("assert");
-mongoose.Promise = global.Promise;
+// const mongoose = require("mongoose");
```
Comment (kanchan-einn) [inline backend/test/AdminModel.test.js:1]: Riya, delete this file if not in use.

File: backend/test/College.test.js [hunk 6]
```diff
@@ -1,134 +1,203 @@
...
-                email:'singh@email.com',
-                password:'rsrsrs',
-                phone:'9494949497',
-                code: '223',
-                address: "Avantika",
-                designation: "TPO"
-            })
-        .expect(400);
-        expect(response.text).to.equal('"designation" is not allowed');
+      console.log("your error", error);
```
Comment (kanchan-einn) [inline backend/test/College.test.js:20]: log proper error message to log file instead on console.

File: backend/test/CollegeModel.test.js [hunk 7]
```diff
@@ -1,139 +1,139 @@
-const mongoose = require("mongoose");
-const Colleges = require("../model/College");
-const assert = require("assert");
-mongoose.Promise = global.Promise;
+// const mongoose = require("mongoose");
```
Comment (kanchan-einn) [inline backend/test/CollegeModel.test.js:1]: Remove this file if not in use.

**Comment by kanchan-einn** [comment]
@Riya-Singhal15 lot of files you have added commented code. Why?

File: backend/controller/errorHandling.js (hunk 3 above)
Comment (Riya-Singhal15) [reply backend/controller/errorHandling.js:200]: When tpo provide a file, he should also provide his email.

File: backend/controller/errorHandling.js (hunk 4 above)
Comment (Riya-Singhal15) [reply backend/controller/errorHandling.js:314]: Okay, I'll update it.

File: backend/test/AdminModel.test.js (hunk 5 above)
Comment (Riya-Singhal15) [reply backend/test/AdminModel.test.js:1]: They are in use when complete test cases will be implemented. I can remove the comments from it.

**Comment by Riya-Singhal15** [comment]
Because those files have code (test cases) that is not working properly right now but I have to fix it after that those files will be committed

File: backend/test/College.test.js [hunk 8]
```diff
@@ -1,134 +1,203 @@
...
-before((done) =>{
-    mongoose.connect("mongodb://localhost/TestingAPIs", {useNewUrlParser: true, useUnifiedTopology: true });
-    mongoose.connection
-       .once('open', () => {
-         console.log("started");
-          done();  
-       })    
-       .on('error', (error) => {
+before((done) => {
+  mongoose.connect("mongodb://localhost/Testing", {
```
Comment (kanchan-einn) [inline backend/test/College.test.js:9]: URL should come from config instead of Hardcoding here.

File: backend/test/AdminAPI.test.js (hunk 1 above)
Comment (Riya-Singhal15) [reply backend/test/AdminAPI.test.js]: Commented files got removed and committed ( commit no. - 26dda2830d863a97de02d7cf0dccfd74cb1047c6).

File: backend/controller/errorHandling.js (hunk 4 above)
Comment (Riya-Singhal15) [reply backend/controller/errorHandling.js]: The message is changed and committed ( commit no. - 26dda2830d863a97de02d7cf0dccfd74cb1047c6).

File: backend/test/AdminModel.test.js (hunk 5 above)
Comment (Riya-Singhal15) [reply backend/test/AdminModel.test.js]: Commented files got removed and committed ( commit no. - 26dda28).

File: backend/test/CollegeModel.test.js (hunk 7 above)
Comment (Riya-Singhal15) [reply backend/test/CollegeModel.test.js]: Commented files got removed and committed ( commit no. - 26dda28).

File: backend/test/College.test.js (hunk 6 above)
Comment (Riya-Singhal15) [reply backend/test/College.test.js]: I have changed from console to logger file inside all test files and committed (commit no. - a210539cf32596b1e31bbd50d014afcda368448a)

File: backend/test/College.test.js (hunk 8 above)
Comment (Riya-Singhal15) [reply backend/test/College.test.js]: I have changed it and put it in .env file and committed (commit no. - a210539)

File: backend/package.json (hunk 2 above)
Comment (Riya-Singhal15) [reply backend/package.json:8]: Nidhi told me that I have to commit the file after putting nodemon in my files rather than pm2.

**Pull request merged**